tests/unit/moduleapi/cluster.tcl

```diff
@@ -0,0 +1,101 @@
+
+# make sure the test infra won't use SELECT
+set ::singledb 1
+
+# start three servers
+start_server {overrides {cluster-enabled yes} tags {"cluster modules"}} {
+start_server {overrides {cluster-enabled yes} tags {"cluster modules"}} {
+start_server {overrides {cluster-enabled yes} tags {"cluster modules"}} {
+
+    set node1 [srv 0 client]
+    set node2 [srv -1 client]
+    set node3 [srv -2 client]
+
+    $node1 module load $testmodule
+    $node2 module load $testmodule
+    $node3 module load $testmodule
+
+    test {Create 3 node cluster} {
+        exec src/redis-cli --cluster-yes --cluster create \
+                           127.0.0.1:[srv 0 port] \
+                           127.0.0.1:[srv -1 port] \
+                           127.0.0.1:[srv -2 port]
+    }
+
+    test "Run blocking command on cluster node3" {
+        # key9184688 is mapped to slot 10923 (first slot of node 3)
+        # use DEL command to wait for the cluster to be stable
+        wait_for_condition 1000 50 {
+            ![catch {$node3 del key9184688}]
+        } else {
+            fail "Cluster doesn't stabilize"
+        }
+
+        set node3_rd [redis_deferring_client -2]
+        $node3_rd fsl.bpop key9184688 0
+        $node3_rd flush
+
+        wait_for_condition 50 100 {
+            [s -2 blocked_clients] eq {1}
+        } else {
+            fail "Client not blocked"
+        }
+    }
+
+    test "Perform a Resharding" {
+        exec src/redis-cli --cluster-yes --cluster reshard 127.0.0.1:[srv -2 port] \
+                           --cluster-to [$node1 cluster myid] \
+                           --cluster-from [$node3 cluster myid] \
+                           --cluster-slots 1
+    }
+
+    test "Verify command got unblocked after resharding" {
+        # this (read) will wait for the node3 to realize the new topology
+        assert_error {*MOVED*} {$node3_rd read}
+
+        # verify there are no blocked clients
+        assert_equal [s 0 blocked_clients]  {0}
+        assert_equal [s -1 blocked_clients]  {0}
+        assert_equal [s -2 blocked_clients]  {0}
+    }
+
+    test "wait for cluster to be stable" {
+        # use EXISTS command to wait for the cluster to be stable
+        wait_for_condition 1000 50 {
+            ![catch {$node1 exists key9184688}]
+        } else {
+            fail "Cluster doesn't stabilize"
+        }
+    }
+
+    test "Sanity test push cmd after resharding" {
+        assert_error {*MOVED*} {$node3 fsl.push key9184688 1}
+
+        set node1_rd [redis_deferring_client 0]
+        $node1_rd fsl.bpop key9184688 0
+        $node1_rd flush
+
+        wait_for_condition 50 100 {
+            [s 0 blocked_clients] eq {1}
+        } else {
+            puts "Client not blocked"
+            puts "read from blocked client: [$node1_rd read]"
+            fail "Client not blocked"
+        }
+
+        $node1 fsl.push key9184688 2
+        assert_equal {2} [$node1_rd read]
+    }
```

Comment:
ok. my test didn't really check that the MOVED response is the correct one, just verified that we do get MOVED, but i think that's enough.. we probably have other tests that check the cluster handles MOVED correctly and maybe even test that redis-cli can follow, **but that's not specifically related to blocked clients or modules, so i think we can skip that**.

I added this sanity test on a simple blocked client so that i'm sure that the other test code (the one that tests resharding) can be trusted. i think i rather keep it.

p.s. we now have two `cluster.tcl` (one for modules and one without), and i'm not sure we should keep them.
my approach was to first write the non-module one which was easier to write and make sure i got it correctly, before i modify it to use the module, but maybe we don't need it now.
however, i have a feeling that sooner or later we'll want to extend this unit (cluster tests in the non-cluster tests) with more tests, so maybe it's a good idea to keep that file there with it's basic infra.